contrib/example-backend/controllers/servicebindingrequest/servicebindingresource_reconcile.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2022 The Kube Bind Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package servicebindingrequest
+
+import (
+	"context"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+
+	kubebindv1alpha1 "github.com/kube-bind/kube-bind/pkg/apis/kubebind/v1alpha1"
+	conditionsapi "github.com/kube-bind/kube-bind/pkg/apis/third_party/conditions/apis/conditions/v1alpha1"
+	"github.com/kube-bind/kube-bind/pkg/apis/third_party/conditions/util/conditions"
+)
+
+type reconciler struct {
+	deleteServiceBindingRequest func(ctx context.Context, ns, name string) error
+
+	getServiceExport    func(ns, name string) (*kubebindv1alpha1.APIServiceExport, error)
+	createServiceExport func(ctx context.Context, resource *kubebindv1alpha1.APIServiceExport) (*kubebindv1alpha1.APIServiceExport, error)
+}
+
+func (r *reconciler) reconcile(ctx context.Context, req *kubebindv1alpha1.APIServiceBindingRequest) error {
+	var errs []error
+
+	if err := r.ensureExports(ctx, req); err != nil {
+		errs = append(errs, err)
+	}
+
+	conditions.SetSummary(req)
+
+	return utilerrors.NewAggregate(errs)
+}
+
+func (r *reconciler) ensureExports(ctx context.Context, req *kubebindv1alpha1.APIServiceBindingRequest) error {
+	logger := klog.FromContext(ctx)
+
+	if req.Status.Export == "" {
+		// create new export
+		export := &kubebindv1alpha1.APIServiceExport{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: req.Namespace,
+				Name:      req.Name,
+			},
+			Spec: kubebindv1alpha1.APIServiceExportSpec{
+				Scope: kubebindv1alpha1.ClusterScope, // TODO: implement namespace scope
+			},
+		}
+		for _, res := range req.Spec.Resources {
+			export.Spec.Resources = append(export.Spec.Resources, kubebindv1alpha1.APIServiceExportGroupResource{
+				GroupResource: res.GroupResource,
+			})
+		}
+		logger.V(2).Info("Creating service export", "namespace", export.Namespace, "name", export.Name)
+		var err error
+		if _, err = r.createServiceExport(ctx, export); err != nil && !apierrors.IsAlreadyExists(err) {
+			export.GenerateName = req.Name + "-"
+			export.Name = ""
+
+			logger.V(2).Info("Creation of service export failed, trying with a generated name", "namespace", export.Namespace, "name", export.Name)
```

Comment:
what's the benefit of this?

Reply:
i.e. vs. always creating with a generate name

Reply:
The APIServiceBindingRequest has ann optional name in the authn response. This is supposed to be used if the service provider keeps hygiene in cluster namespaces. But this cannot be guaranteed. Hence, this special case.